admin-tools/lambda/src/main/scala/com/gu/mediaservice/BatchIndexHandlerAwsFunctions.scala

```diff
@@ -0,0 +1,88 @@
+package com.gu.mediaservice
+
+import java.io.ByteArrayInputStream
+import java.nio.ByteBuffer
+import java.util.UUID
+
+import com.amazonaws.client.builder.AwsClientBuilder.EndpointConfiguration
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient
+import com.amazonaws.services.dynamodbv2.document.{Table, DynamoDB => AwsDynamoDB}
+import com.amazonaws.services.kinesis.model.PutRecordRequest
+import com.amazonaws.services.kinesis.{AmazonKinesis, AmazonKinesisClientBuilder}
+import com.amazonaws.services.s3.AmazonS3ClientBuilder
+import com.amazonaws.services.s3.model.ObjectMetadata
+import com.gu.mediaservice.lib.json.JsonByteArrayUtil
+import play.api.libs.json.JsValue
+
+class BatchIndexHandlerAwsFunctions(cfg: BatchIndexHandlerConfig) {
+
+  private val AwsRegion = "eu-west-1"
+  private val s3client = buildS3Client
+  private val kinesis = buildKinesisClient
+
+  import cfg._
+
+  def putToS3(imageBlobs: List[String]): String = {
+    val fileContent = imageBlobs.mkString("\n")
+    val key = s"batch-index/${UUID.randomUUID().toString}.json"
+    val metadata = new ObjectMetadata
+    metadata.setContentType("application/json")
+    val res = s3client.putObject(batchIndexBucket, key, new ByteArrayInputStream(fileContent.getBytes), metadata)
+    val path = s"s3://$batchIndexBucket/$key"
```

Comment:
I think the message just needs to be the key as `thrall` should get the bucket from config.

Reply:
i will prefer to provide that in message rather that introducing another config to be honest

Reply:
or we can use case class for S3Location(bucket: String, key: String) if thrall s3 client needs that

Reply:
i mention that in your PR: https://github.com/guardian/grid/pull/2767/files#diff-9e12ba48c7d2af2248094256c01c3d36R100

Reply:
refined